book.rb

```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental) unless @rentals.include?(rental)
+    rental.book = self
+  end
```

Comment:
When you created the `rental` class, in the `constructor`, you add the rental object to both the `person`'s rentals array and the `book`'s rentals array via the following lines: 
```
 person.rentals << self
 book.rentals << self
```
 So, it would make more sense if instead of passing a rental object as a parameter here, you pass the `date` and the `person`, and in the body of the method, you create a new rental (To create a new rental, we also need a `book`, for that, you can use `self`).  This way, the constructor will take care of adding it to the list of rentals for both the book and the person.

Something like this will do the trick:
```
   def add_rental(book, date)
       Rental.new(date, self, person)
   end

```